app/Phragile/TransactionRawDataProcessor.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Phragile;
+
+class TransactionRawDataProcessor {
+
+	public function process(array $rawData)
+	{
+		$transactions = [];
+		foreach ($rawData as $task => $taskTransactionData)
+		{
+			$transactions[$task] = $this->processTaskTransactionData($taskTransactionData);
+		}
+		return $transactions;
+	}
+
+	private function processTaskTransactionData(array $rawTaskTransactionData)
+	{
+		$transactions = [];
+		foreach ($rawTaskTransactionData as $singleTransactionData)
+		{
+			if ($this->isColumnChangeTransaction($singleTransactionData))
+			{
+				$transactions[] = new ColumnChangeTransaction([
+					'timestamp' => $this->getTransactionTimestamp($singleTransactionData),
+					'workboardPHID' => $this->getWorkboardPHID($singleTransactionData),
+					'oldColumnPHID' => $this->getOldColumnPHID($singleTransactionData),
+					'newColumnPHID' => $this->getNewColumnPHID($singleTransactionData),
+				]);
+			} elseif ($this->isStatusChangeTransaction($singleTransactionData))
+			{
+				$transactions[] = new StatusChangeTransaction([
+					'timestamp' => $this->getTransactionTimestamp($singleTransactionData),
+					'oldStatus' => $this->getOldStatus($singleTransactionData),
+					'newStatus' => $this->getNewStatus($singleTransactionData),
+				]);
+			} elseif ($this->isMergedIntoTransaction($singleTransactionData))
+			{
+				$transactions[] = new MergedIntoTransaction(
+					$this->getTransactionTimestamp($singleTransactionData)
+				);
+			}
+		}
+		return $transactions;
+	}
+
+	private function isColumnChangeTransaction(array $rawData)
+	{
+		return array_key_exists('transactionType', $rawData)
+			&& $rawData['transactionType'] === 'core:columns';
```

Comment:
I don't think this belongs to Transaction class, it is Phabricator-specific.
I think this is only line of production code that mentions `core:columns` (some test data might be also doing it which is expected I think). And if not only then only Phabricator2Phragile transaction data conversion code should know anything on how currently Phabricator represents column change transactions (and any other kind of transactions).
Any way, be it he only line explicitly mentioning `core:columns` it might make sense to make those Phabricator transaction type string class constants so they are changed in one place no matter how the processor's code evolves. At least I think it might help a bit, so I'll change it. Of course it would only be valuable unless next change of transaction data structure in Phab API will be making it all look completely differently but this is rather unlikely.